<a id="top"></a>
# Hackathon Notebook: Early Childhood development 
### Team 3: Explore DS Academy
---
<img src="https://images.ctfassets.net/9l3tjzgyn9gr/6DNCGMxB4N77zOk5a7y7JK/b03781ef3ac46123874b51bb3b9c4e75/education-and-child-development.jpg?fm=jpg&fl=progressive&q=50&w=1200" align="left">

**Team Members:** David Kambo, Justice Oyemike, Oludare Adekunle, Okon Prince  
**Project Manager:** Junita Henry

* [Notebook repo](https://github.com/Project/Team_3)
* [Trello board](https://trello.com/b/1xeVg7gj/team_3) 

<a id="intro"></a>
## 1. Introduction  

### Background

The size of a child’s brain reaches 90 per cent of an adult’s by the age of five. Early childhood is a time of rapid change – particularly for the development of a child’s brain. These early years are critical for lifelong learning and well-being. Life long social and cognitive skills such as talking, reading, playing logical reasoning capacity are developed at this stage. Kids also develop social skills like empathy, friendship and co-operation at this stage of their lives. Asides from their homes. The school environment is the place outside a childs home where they spend the most time. Moreso, children get to spend time with their peers in the school environment creating the ideal opportunity for these skills and bonds to form. The early years are very important in shaping the cognitive and emotional patterns of kids for life, thus influencing learning, relationships and resilience permanently.

However, the importance of this development and learning process, The education of the pre-school child has not been given sufficient attention by the athorities.  Decision-makers continue to treat our early learning and childcare system as a child-minding service for parents returning to work. There is an erroneous notion that kids are too young to be taken seriously with a tendency to take child education as serious business only after the child communication abilities are well established. This is a grave mistake as most times, the opportunity to mould the kid and influence future capacities might have been lost. 

It’s time the value of play-based learning is recognised by making early learning system high-quality, universally accessible and attached to the public education system so that every child can benefit.

### Problem Statement
Construct a Machine learning algorithm that takes the data collected from pre-school children and develops a Model that can accurately predict the likelihood of a child to grow into an adult who will do well in school.

### Revelations
By this project, we hope to reveal the immediate and remote factors that influences positive deviance in in academics and ...

## Import Dependencies
Import all the dependencies that is required for the work

In [4]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.feature_selection import VarianceThreshold

## Prepare the Data
### Load the Data set
Load the stata file data into a pandas data frame

In [5]:
# Load dataset
df = pd.read_stata("DataDrive2030_PD_EXTERNAL.dta")

In [6]:
pd.set_option('display.max_columns', None)

In [12]:
print("df shape:", df.shape)
df.head(1)

df shape: (11822, 1288)


child_id  data_year           data_source child_date  child_age  \
0  1000101     2021.0  Thrive by Five Index 2021-10-18       59.0   

  child_enrolment_date  child_months_enrolment child_grant  \
0           2020-01-13                    21.0         NaN   

  child_years_in_programme  child_score_item_1  child_score_item_2  \
0    2nd year in programme                2.91                2.07   

   child_score_item_3  child_score_item_4  child_domain_1  child_score_item_5  \
0                 0.0                4.25            9.23                 4.2   

   child_score_item_6  child_score_item_7  child_score_item_8  child_domain_2  \
0                4.58                3.77                4.51       17.059999   

   child_score_item_9  child_score_item_10  child_score_item_11  \
0                2.81                 2.84                 2.47   

   child_score_item_12  child_score_item_13  child_domain_3  \
0                 1.38                 2.65           12.15   

   child_score_item_14  child_score_item_15  child_score_item_16  \
0                 3.11                 1.34                 4.03   

   child_score_item_17  child_domain_4  child_score_item_18  \
0                 6.14       14.619999                 3.21   

   child_score_item_19  child_score_item_20  child_score_item_21  \
0                  0.0                 2.47                  2.4   

   child_score_item_22  child_score_item_23  child_domain_5  child_height  \
0                 2.19                 2.97           13.24    114.800003   

  child_observe_attentive child_observe_concentrated child_observe_diligent  \
0                   Often                      Often                  Often   

  child_observe_interested  child_observe_total  child_total_elom  \
0                    Often                  8.0         66.300003   

  child_total_elom_cuts child_domain_1_cuts child_domain_2_cuts  \
0              On track            On track            On track   

  child_domain_3_cuts child_domain_4_cuts child_domain_5_cuts child_gender  \
0            On track            On track            On track       Female   

   child_sample_wgt  child_dob  child_age_w2 child_date_w2  \
0           224.125 2016-11-09           NaN           NaT   

   child_score_item_1_w2  child_score_item_2_w2  child_score_item_3_w2  \
0                    NaN                    NaN                    NaN   

   child_score_item_4_w2  child_domain_1_w2  child_score_item_5_w2  \
0                    NaN                NaN                    NaN   

   child_score_item_6_w2  child_score_item_7_w2  child_score_item_8_w2  \
0                    NaN                    NaN                    NaN   

   child_domain_2_w2  child_score_item_9_w2  child_score_item_10_w2  \
0                NaN                    NaN                     NaN   

   child_score_item_11_w2  child_score_item_12_w2  child_score_item_13_w2  \
0                     NaN                     NaN                     NaN   

   child_domain_3_w2  child_score_item_14_w2  child_score_item_15_w2  \
0                NaN                     NaN                     NaN   

   child_score_item_16_w2  child_score_item_17_w2  child_domain_4_w2  \
0                     NaN                     NaN                NaN   

   child_score_item_18_w2  child_score_item_19_w2  child_score_item_20_w2  \
0                     NaN                     NaN                     NaN   

   child_score_item_21_w2  child_score_item_22_w2  child_score_item_23_w2  \
0                     NaN                     NaN                     NaN   

   child_domain_5_w2  child_total_elom_w2  child_height_w2  \
0                NaN                  NaN              NaN   

  child_observe_attentive_w2 child_observe_concentrated_w2  \
0                        NaN                           NaN   

  child_observe_diligent_w2 child_observe_interested_w2  \
0                       NaN                         NaN   

   child_observe_total_w2 child_

One of the first things you might notice here is that this dataset is Large — 11,822 rows and 1288 columns to be precise! SO MUCH DATA!!! We won't have time to explore all of the features in this dataset, but you can look in the [data dictionary](./066-data-dictionary.ipynb) for this project for details and links to the official [Code Book](https://sda.berkeley.edu/sdaweb/docs/scfcomb2019/DOC/hcbk.htm). For now, let's just say that this dataset tracks all sorts of behaviors relating to the dynamics that affect how people live, where they live, the educational choices for their kids, how much they spend on kids education etc in South Africa.

For this project, we will focus on the indices that reveal the performance of Kids in the schools.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 1288 entries, child_id to id_prov_n
dtypes: category(808), datetime64[ns](12), float32(126), float64(232), object(110)
memory usage: 46.9+ MB


In [10]:
df.info

<bound method DataFrame.info of       child_id  data_year           data_source child_date  child_age  \
0      1000101     2021.0  Thrive by Five Index 2021-10-18   59.00000   
1      1000102     2021.0  Thrive by Five Index 2021-10-18   58.00000   
2      1000103     2021.0  Thrive by Five Index 2021-10-18   58.00000   
3      1000104     2021.0  Thrive by Five Index 2021-10-18   57.00000   
4      1000202     2021.0  Thrive by Five Index 2021-10-25   54.00000   
...        ...        ...                   ...        ...        ...   
11817  1049704     2021.0  Thrive by Five Index 2021-09-21   55.00000   
11818  1054602     2021.0  Thrive by Five Index 2021-10-12   56.00000   
11819  1107805     2019.0             DataFirst        NaT   57.80328   
11820  1058601     2021.0  Thrive by Five Index 2021-11-11   53.00000   
11821  1042201     2022.0                 Other 2022-03-15   62.00000   

      child_enrolment_date  child_months_enrolment child_grant  \
0               2020-01-1

In [3]:
var_thres=VarianceThreshold(threshold=0)
var_thres.fit(df)

NameError: name 'VarianceThreshold' is not defined

In [11]:
# create OBS Subset data 
scores_df= data[['child_score_item_1', 'child_score_item_2', 'child_score_item_3',
       'child_score_item_4', 'child_score_item_5',
       'child_score_item_6', 'child_score_item_7', 'child_score_item_8',
       'child_score_item_9', 'child_score_item_10',
       'child_score_item_11', 'child_score_item_12', 'child_score_item_13',
       'child_score_item_14', 'child_score_item_15',
       'child_score_item_16', 'child_score_item_17', 
       'child_score_item_18', 'child_score_item_19', 'child_score_item_20',
       'child_score_item_21', 'child_score_item_22', 'child_score_item_23',
       ]]

domain_df= data[[
    'child_domain_1', 'child_domain_2', 'child_domain_3', 
    'child_domain_4', 'child_domain_5'
]]

In [27]:
subset_list= []
for i in df.columns:
    if 'pqa_' in i:
        subset_list.append(i)
        
pqa_subset= df[subset_list]

In [28]:
pqa_subset.head()

pqa_date pqa_class_age pqa_class_age_1 pqa_class_age_2 pqa_class_age_3  \
0      NaT                           NaN             NaN             NaN   
1      NaT                           NaN             NaN             NaN   
2      NaT                           NaN             NaN             NaN   
3      NaT                           NaN             NaN             NaN   
4      NaT                           NaN             NaN             NaN   

  pqa_class_age_4 pqa_class_age_5 pqa_class_age_6  pqa_class_assistants  \
0             NaN             NaN             NaN                   NaN   
1             NaN             NaN             NaN                   NaN   
2             NaN             NaN             NaN                   NaN   
3             NaN             NaN             NaN                   NaN   
4             NaN             NaN             NaN                   NaN   

  pqa_environment_areas pqa_environment_variety pqa_environment_appropriate  \
0                   NaN                     NaN                         NaN   
1                   NaN                     NaN                         NaN   
2                   NaN                     NaN                         NaN   
3                   NaN                     NaN                         NaN   
4                   NaN                     NaN                         NaN   

  pqa_environment_accessible pqa_environment_open pqa_environment_outdoor  \
0                        NaN                  NaN                     NaN   
1                        NaN                  NaN                     NaN   
2                        NaN                  NaN                     NaN   
3                        NaN                  NaN                     NaN   
4                        NaN                  NaN                     NaN   

   pqa_score_environment pqa_assessment_observation pqa_assessment_systematic  \
0                    NaN                        NaN                       NaN   
1                    NaN                        NaN                       NaN   
2                    NaN                        NaN                       NaN   
3                    NaN                        NaN                       NaN   
4                    NaN                        NaN                       NaN   

   pqa_score_assessment pqa_relationships_peers pqa_relationships_staff  \
0                   NaN                     NaN                     NaN   
1                   NaN                     NaN                     NaN   
2                   NaN                     NaN                     NaN   
3                   NaN                     NaN                     NaN   
4                   NaN                     NaN                     NaN   

  pqa_relationships_acknowledge pqa_relationships_discipline  \
0                           NaN                          NaN   
1                           NaN                          NaN   
2                           NaN                          NaN   
3                           NaN                          NaN   
4                           NaN                          NaN   

   pqa_score_relationships pqa_curriculum_ncf pqa_curriculum_plan  \
0                      NaN                NaN                 NaN   
1                      NaN                NaN                 NaN   
2                      NaN                NaN                 NaN   
3                      NaN                NaN                 NaN   
4                      NaN                NaN                 NaN   

  pqa_curriculum_balance pqa_curriculum_numeracy pqa_curriculum_literacy  \
0                    NaN                     NaN                     NaN   
1                    NaN                     NaN                     NaN   
2                    NaN                     NaN                     NaN   
3                    NaN                     NaN                     NaN   
4                    NaN              

In [30]:
pqa_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Data columns (total 37 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   pqa_date                       2221 non-null   datetime64[ns]
 1   pqa_class_age                  11822 non-null  object        
 2   pqa_class_age_1                2221 non-null   category      
 3   pqa_class_age_2                2221 non-null   category      
 4   pqa_class_age_3                2221 non-null   category      
 5   pqa_class_age_4                2221 non-null   category      
 6   pqa_class_age_5                2221 non-null   category      
 7   pqa_class_age_6                2221 non-null   category      
 8   pqa_class_assistants           2221 non-null   float64       
 9   pqa_environment_areas          2221 non-null   category      
 10  pqa_environment_variety        2221 non-null   category      
 11  pqa_environment

In [25]:
subset_list= []
for i in df.columns:
    if 'child_' in i:
        subset_list.append(i)
        
child_subset= df[subset_list]

In [26]:
child_subset.head()

child_id child_date  child_age child_enrolment_date  child_months_enrolment  \
0  1000101 2021-10-18       59.0           2020-01-13                    21.0   
1  1000102 2021-10-18       58.0           2020-01-18                    21.0   
2  1000103 2021-10-18       58.0           2020-02-15                    20.0   
3  1000104 2021-10-18       57.0           2020-02-18                    19.0   
4  1000202 2021-10-25       54.0           2021-02-01                     8.0   

  child_grant   child_years_in_programme  child_score_item_1  \
0         NaN      2nd year in programme                2.91   
1         NaN      2nd year in programme                2.91   
2         NaN      2nd year in programme                2.91   
3         NaN      2nd year in programme                2.91   
4         Yes  1st year in the programme                2.91   

   child_score_item_2  child_score_item_3  child_score_item_4  child_domain_1  \
0                2.07                0.00                4.25            9.23   
1                1.00                2.22                4.25           10.38   
2                2.07                0.00                2.62            7.60   
3                0.00                2.22                0.00            5.13   
4                3.59                2.22                2.62           11.34   

   child_score_item_5  child_score_item_6  child_score_item_7  \
0                4.20                4.58                3.77   
1                4.20                4.58                4.74   
2                3.11                0.00                3.77   
3                4.20                4.58                3.12   
4                3.11                0.00                3.77   

   child_score_item_8  child_domain_2  child_score_item_9  \
0                4.51       17.059999                2.81   
1                5.33       18.850000                4.05   
2                3.95       10.830000                0.00   
3                3.41       15.310000                2.81   
4                2.75        9.630000                2.81   

   child_score_item_10  child_score_item_11  child_score_item_12  \
0                 2.84                 2.47                 1.38   
1                 2.84                 2.47                 2.80   
2                 0.00                 2.47                 1.38   
3                 2.84                 2.47                 2.80   
4                 0.00                 2.47                 1.38   

   child_score_item_13  child_domain_3  child_score_item_14  \
0                 2.65           12.15                 3.11   
1                 2.65           14.81                 3.11   
2                 1.00            4.85                 1.00   
3                 2.65           13.57                 3.11   
4                 2.65            9.31                 3.11   

   child_score_item_15  child_score_item_16  child_score_item_17  \
0                 1.34                 4.03                 6.14   
1                 1.34                 4.03                 3.02   
2                 0.00                 4.03                 6.14   
3                 0.00                 2.11                 6.14   
4                 0.00                 4.03                 0.00   

   child_domain_4  child_score_item_18  child_score_item_19  \
0       14.619999                 3.21                 0.00   
1       11.500000                 3.21                 2.64   
2       11.170000                 0.00                 2.64   
3       11.360000                 3.21                 2.64   
4        7.140000                 0.00                 0.00   

   child_score_item_20  child_score_item_21  child_score_item_22  \
0                 2.47                 2.40                 2.19   
1                 2.47                 2.40                 3.09   
2                 2.47                 2.40                 2.19   
3                 2.47               

In [31]:
child_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Data columns (total 98 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   child_id                       11822 non-null  object        
 1   child_date                     8955 non-null   datetime64[ns]
 2   child_age                      11822 non-null  float64       
 3   child_enrolment_date           3574 non-null   datetime64[ns]
 4   child_months_enrolment         3574 non-null   float64       
 5   child_grant                    7374 non-null   category      
 6   child_years_in_programme       9574 non-null   category      
 7   child_score_item_1             11822 non-null  float64       
 8   child_score_item_2             11822 non-null  float64       
 9   child_score_item_3             11822 non-null  float64       
 10  child_score_item_4             11822 non-null  float64       
 11  child_domain_1 

In [21]:
subset_list= []
for i in df.columns:
    if 'pri_' in i:
        subset_list.append(i)
        
pri_subset= df[subset_list]

In [22]:
pri_subset.head()

pri_mobile pri_school pri_holidays  pri_days  \
0  Based at a specific location         No           No       5.0   
1  Based at a specific location         No           No       5.0   
2  Based at a specific location         No           No       5.0   
3  Based at a specific location         No           No       5.0   
4  Based at a specific location         No          Yes       5.0   

  pri_time_open_hours pri_time_open_minutes pri_time_close_hours  \
0                  07                    00                   14   
1                  07                    00                   14   
2                  07                    00                   14   
3                  07                    00                   14   
4                  07                    00                   16   

  pri_time_close_minutes pri_calc_time_open pri_calc_time_close  pri_year  \
0                     00              07:00               14:00    1993.0   
1                     00              07:00               14:00    1993.0   
2                     00              07:00               14:00    1993.0   
3                     00              07:00               14:00    1993.0   
4                     00              07:00               16:00    1980.0   

                                        pri_separate pri_language  \
0  Yes, children are grouped by age but are using...            2   
1  Yes, children are grouped by age but are using...            2   
2  Yes, children are grouped by age but are using...            2   
3  Yes, children are grouped by age but are using...            2   
4  Yes, children are grouped by age but are using...          2 5   

  pri_language_1 pri_language_2 pri_language_3 pri_language_4 pri_language_5  \
0             No            Yes             No             No             No   
1             No            Yes             No             No             No   
2             No            Yes             No             No             No   
3             No            Yes             No             No             No   
4             No            Yes             No             No            Yes   

  pri_language_6 pri_language_7 pri_language_8 pri_language_9 pri_language_10  \
0             No             No             No             No              No   
1             No             No             No             No              No   
2             No             No             No             No              No   
3             No             No             No             No              No   
4             No             No             No             No              No   

  pri_language_11 pri_language_97 pri_languageother pri_toys pri_aftercare  \
0              No              No                         No            No   
1              No              No                         No            No   
2              No              No                         No            No   
3              No              No                         No            No   
4              No              No                         No           Yes   

   pri_toy_members_orgs  pri_toy_members_individuals  pri_toy_count pri_fees  \
0                   NaN                          NaN            NaN      Yes   
1                   NaN                          NaN            NaN      Yes   
2                   NaN                          NaN            NaN      Yes   
3                   NaN                          NaN            NaN      Yes   
4                   NaN                          NaN            NaN      Yes   

   pri_fees_amount pri_fees_free pri_facilities      pri_land  \
0            250.0            No   Municipality  Municipality   
1            250.0            No   Municipality  Municipality   
2            250.0            No   Municipality  Municipality   
3            250.0            No   Municipality  Municipality   
4            350.0           Yes   Municipality  Municipality   

  pri_faci

In [24]:
pri_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 304 entries, pri_mobile to saspri_datazone_quintile
dtypes: category(213), datetime64[ns](1), float32(2), float64(43), object(45)
memory usage: 10.6+ MB


In [35]:
subset_list= []
for i in df.columns:
    if 'pra_' in i:
        subset_list.append(i)
        
pra_subset= df[subset_list]

In [36]:
pra_subset.head()

pra_free_play pra_free_play_outdoor pra_groupings pra_groupings_1  \
0  30 minutes or less    30 minutes or less             5              No   
1  30 minutes or less    30 minutes or less             5              No   
2  30 minutes or less    30 minutes or less             5              No   
3  30 minutes or less    30 minutes or less             5              No   
4  30 minutes or less    30 minutes or less             5              No   

  pra_groupings_2 pra_groupings_3 pra_groupings_4 pra_groupings_5 pra_engaged  \
0              No              No              No             Yes       Often   
1              No              No              No             Yes       Often   
2              No              No              No             Yes       Often   
3              No              No              No             Yes       Often   
4              No              No              No             Yes       Often   

  pra_plans pra_plans_1 pra_plans_2 pra_plans_3 pra_plans_0 pra_gender  \
0     1 2 3         Yes         Yes         Yes          No        NaN   
1     1 2 3         Yes         Yes         Yes          No        NaN   
2     1 2 3         Yes         Yes         Yes          No        NaN   
3     1 2 3         Yes         Yes         Yes          No        NaN   
4     1 2 3         Yes         Yes         Yes          No        NaN   

   pra_experience  pra_class_present pra_date  pra_class_size  \
0             NaN                NaN      NaT             NaN   
1             NaN                NaN      NaT             NaN   
2             NaN                NaN      NaT             NaN   
3             NaN                NaN      NaT             NaN   
4             NaN                NaN      NaT             NaN   

   pra_class_attendance  pra_class_attendance_precovid pra_class_size_large  \
0                   NaN                            NaN                        
1                   NaN                            NaN                        
2                   NaN                            NaN                        
3                   NaN                            NaN                        
4                   NaN                            NaN                        

  pra_cohort pra_cohort_0 pra_cohort_1 pra_cohort_2 pra_cohort_3 pra_cohort_4  \
0                     NaN          NaN          NaN          NaN          NaN   
1                     NaN          NaN          NaN          NaN          NaN   
2                     NaN          NaN          NaN          NaN          NaN   
3                     NaN          NaN          NaN          NaN          NaN   
4                     NaN          NaN          NaN          NaN          NaN   

  pra_cohort_5 pra_cohort_6 pra_class_language pra_plan_4yrs pra_plan_4yrs_1  \
0          NaN          NaN                NaN                           NaN   
1          NaN          NaN                NaN                           NaN   
2          NaN          NaN                NaN                           NaN   
3          NaN          NaN                NaN                           NaN   
4          NaN          NaN                NaN                           NaN   

  pra_plan_4yrs_2 pra_plan_4yrs_3 pra_plan_4yrs_4 pra_plan_4yrs_97  \
0             NaN             NaN             NaN              NaN   
1             NaN             NaN             NaN              NaN   
2             NaN             NaN             NaN              NaN   
3             NaN             NaN             NaN              NaN   
4             NaN             NaN             NaN              NaN   

  pra_plan_ncf pra_plan_5yrs pra_plan_5yrs_1 pra_plan_5yrs_2 pra_plan_5yrs_3  \
0          NaN                           NaN             NaN             NaN   
1          NaN                           NaN             NaN             NaN   
2          NaN                           NaN             NaN             NaN   
3          NaN                        

In [37]:
pra_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 152 entries, pra_free_play to pra_language
dtypes: category(111), datetime64[ns](2), float32(5), float64(15), object(19)
memory usage: 4.8+ MB


In [39]:
subset_list= []
for i in df.columns:
    if 'obs_' in i:
        subset_list.append(i)
        
obs_subset= df[subset_list]

In [40]:
obs_subset.head()

obs_firstaid  obs_classrooms obs_space         obs_area obs_area_1  \
0          Yes             3.0       Yes      1 2 3 4 6 7        Yes   
1          Yes             3.0       Yes      1 2 3 4 6 7        Yes   
2          Yes             3.0       Yes      1 2 3 4 6 7        Yes   
3          Yes             3.0       Yes      1 2 3 4 6 7        Yes   
4          Yes             8.0       Yes  1 2 3 4 5 6 7 8        Yes   

  obs_area_2 obs_area_3 obs_area_4 obs_area_5 obs_area_6 obs_area_7  \
0        Yes        Yes        Yes         No        Yes        Yes   
1        Yes        Yes        Yes         No        Yes        Yes   
2        Yes        Yes        Yes         No        Yes        Yes   
3        Yes        Yes        Yes         No        Yes        Yes   
4        Yes        Yes        Yes        Yes        Yes        Yes   

  obs_area_8 obs_area_0                                      obs_materials  \
0         No         No  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 1...   
1         No         No  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 1...   
2         No         No  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 1...   
3         No         No  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 1...   
4        Yes         No   1 2 3 4 5 7 8 9 10 11 12 13 14 15 16 17 18 19 20   

  obs_materials_1 obs_materials_2 obs_materials_3 obs_materials_4  \
0             Yes             Yes             Yes             Yes   
1             Yes             Yes             Yes             Yes   
2             Yes             Yes             Yes             Yes   
3             Yes             Yes             Yes             Yes   
4             Yes             Yes             Yes             Yes   

  obs_materials_5 obs_materials_6 obs_materials_7 obs_materials_8  \
0             Yes             Yes             Yes             Yes   
1             Yes             Yes             Yes             Yes   
2             Yes             Yes             Yes             Yes   
3             Yes             Yes             Yes             Yes   
4             Yes              No             Yes             Yes   

  obs_materials_9 obs_materials_10 obs_materials_11 obs_materials_12  \
0             Yes              Yes              Yes              Yes   
1             Yes              Yes              Yes              Yes   
2             Yes              Yes              Yes              Yes   
3             Yes              Yes              Yes              Yes   
4             Yes              Yes              Yes              Yes   

  obs_materials_13 obs_materials_14 obs_materials_15 obs_materials_16  \
0              Yes              Yes              Yes              Yes   
1              Yes              Yes              Yes              Yes   
2              Yes              Yes              Yes              Yes   
3              Yes              Yes              Yes              Yes   
4              Yes              Yes              Yes              Yes   

  obs_materials_17 obs_materials_18 obs_materials_19 obs_materials_20  \
0              Yes              Yes              Yes              Yes   
1              Yes              Yes              Yes              Yes   
2              Yes              Yes              Yes              Yes   
3              Yes              Yes              Yes              Yes   
4              Yes              Yes              Yes              Yes   

  obs_materials_97 obs_materials_0 obs_materialsother obs_accessible  \
0               No              No                               Yes   
1               No              No                               Yes   
2               No              No                               Yes   
3               No              No                               Yes   
4               No              No                               Yes   

  obs_books obs_books_age obs_heating_census obs_lighting_census  \
0       Yes           Yes      

In [41]:
obs_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 178 entries, obs_firstaid to obs_cooking_6
dtypes: category(156), datetime64[ns](1), float64(4), object(17)
memory usage: 3.9+ MB


In [43]:
subset_list= []
for i in df.columns:
    if 'hle_' in i:
        subset_list.append(i)
        
hle_subset= df[subset_list]

In [44]:
hle_subset.head()

hle_date hle_caregiver_dob hle_caregiver_relation  hle_burdenofcare  \
0      NaT               NaT                    NaN               NaN   
1      NaT               NaT                    NaN               NaN   
2      NaT               NaT                    NaN               NaN   
3      NaT               NaT                    NaN               NaN   
4      NaT               NaT                    NaN               NaN   

  hle_caregiver_postschool hle_time_available_weekday  \
0                      NaN                        NaN   
1                      NaN                        NaN   
2                      NaN                        NaN   
3                      NaN                        NaN   
4                      NaN                        NaN   

  hle_time_available_weekend  hle_picturebooks hle_homemade_toys  \
0                        NaN               NaN               NaN   
1                        NaN               NaN               NaN   
2                        NaN               NaN               NaN   
3                        NaN               NaN               NaN   
4                        NaN               NaN               NaN   

  hle_homemade_toys_string hle_shop_toys hle_shop_toys_string  \
0                                    NaN                        
1                                    NaN                        
2                                    NaN                        
3                                    NaN                        
4                                    NaN                        

  hle_objects_as_toys hle_objects_as_toys_string hle_readbooks_time  \
0                 NaN                                           NaN   
1                 NaN                                           NaN   
2                 NaN                                           NaN   
3                 NaN                                           NaN   
4                 NaN                                           NaN   

  hle_readbooks_noone hle_readbooks_mother hle_readbooks_father  \
0                 NaN                  NaN                  NaN   
1                 NaN                  NaN                  NaN   
2                 NaN                  NaN                  NaN   
3                 NaN                  NaN                  NaN   
4                 NaN                  NaN                  NaN   

  hle_readbooks_auntoruncle hle_readbooks_grandparent hle_readbooks_sibling  \
0                       NaN                       NaN                   NaN   
1                       NaN                       NaN                   NaN   
2                       NaN                       NaN                   NaN   
3                       NaN                       NaN                   NaN   
4                       NaN                       NaN                   NaN   

  hle_readbooks_other hle_readbooks_otherwho hle_stories_time  \
0                 NaN                    NaN              NaN   
1                 NaN                    NaN              NaN   
2                 NaN                    NaN              NaN   
3                 NaN                    NaN              NaN   
4                 NaN                    NaN              NaN   

  hle_stories_noone hle_stories_mother hle_stories_father  \
0               NaN                NaN                NaN   
1               NaN                NaN                NaN   
2               NaN                NaN                NaN   
3               NaN                NaN                NaN   
4               NaN                NaN                NaN   

  hle_stories_auntoruncle hle_stories_grandparent hle_stories_sibling  \
0                     NaN                     NaN                 NaN   
1                     NaN                     NaN                 NaN   
2                     NaN                     NaN                 NaN   
3                     NaN                     NaN                

In [45]:
hle_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 116 entries, hle_date to hle_ind
dtypes: category(100), datetime64[ns](2), float64(8), object(6)
memory usage: 2.7+ MB


In [46]:
subset_list= []
for i in df.columns:
    if 'PD_' in i:
        subset_list.append(i)
        
PD_subset= df[subset_list]

In [47]:
PD_subset.head()

PD_1_INT_total_elom_CHILD PD_1_INT_domain_1_CHILD PD_1_INT_domain_2_CHILD  \
0               Non-deviant             Non-deviant             Non-deviant   
1               Non-deviant             Non-deviant             Non-deviant   
2               Non-deviant             Non-deviant             Non-deviant   
3               Non-deviant             Non-deviant             Non-deviant   
4               Non-deviant             Non-deviant             Non-deviant   

  PD_1_INT_domain_3_CHILD PD_1_INT_domain_4_CHILD PD_1_INT_domain_5_CHILD  \
0             Non-deviant             Non-deviant             Non-deviant   
1             Non-deviant             Non-deviant             Non-deviant   
2             Non-deviant             Non-deviant             Non-deviant   
3             Non-deviant             Non-deviant             Non-deviant   
4             Non-deviant             Non-deviant             Non-deviant   

     PD_2_INT_CHILD PD_4_INT_CHILD PD_1_INT_total_elom_facility  \
0  Positive deviant    Non-deviant                  Non-deviant   
1  Positive deviant    Non-deviant                  Non-deviant   
2       Non-deviant    Non-deviant                  Non-deviant   
3       Non-deviant    Non-deviant                  Non-deviant   
4       Non-deviant    Non-deviant                  Non-deviant   

  PD_1_INT_domain_1_facility PD_1_INT_domain_2_facility  \
0                Non-deviant                Non-deviant   
1                Non-deviant                Non-deviant   
2                Non-deviant                Non-deviant   
3                Non-deviant                Non-deviant   
4                Non-deviant                Non-deviant   

  PD_1_INT_domain_3_facility PD_1_INT_domain_4_facility  \
0                Non-deviant                Non-deviant   
1                Non-deviant                Non-deviant   
2                Non-deviant                Non-deviant   
3                Non-deviant                Non-deviant   
4                Non-deviant                Non-deviant   

  PD_1_INT_domain_5_facility PD_2_INT_pc_all_ontrack_facility  \
0                Non-deviant                 Positive deviant   
1                Non-deviant                 Positive deviant   
2                Non-deviant                 Positive deviant   
3                Non-deviant                 Positive deviant   
4                Non-deviant                      Non-deviant   

  PD_3_INT_pc_ontrack_facility PD_4_INT_composite_facility  \
0                  Non-deviant                 Non-deviant   
1                  Non-deviant                 Non-deviant   
2                  Non-deviant                 Non-deviant   
3                  Non-deviant                 Non-deviant   
4                  Non-deviant                 Non-deviant   

  PD_1_INT_total_elom_ward PD_1_INT_domain_1_ward PD_1_INT_domain_2_ward  \
0              Non-deviant            Non-deviant       Positive deviant   
1              Non-deviant            Non-deviant       Positive deviant   
2              Non-deviant            Non-deviant       Positive deviant   
3              Non-deviant            Non-deviant       Positive deviant   
4              Non-deviant            Non-deviant            Non-deviant   

  PD_1_INT_domain_3_ward PD_1_INT_domain_4_ward PD_1_INT_domain_5_ward  \
0            Non-deviant            Non-deviant            Non-deviant   
1            Non-deviant            Non-deviant            Non-deviant   
2            Non-deviant            Non-deviant            Non-deviant   
3            Non-deviant            Non-deviant            Non-deviant   
4            Non-deviant            Non-deviant            Non-deviant   

  PD_2_INT_pc_all_ontrack_ward PD_3_INT_pc_ontrack_ward  \
0                  Non-deviant              Non-deviant   
1                  Non-deviant              Non-deviant   
2                  Non-deviant              Non-deviant   
3                  Non-deviant              Non-d

In [48]:
PD_subset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11822 entries, 0 to 11821
Columns: 136 entries, PD_1_INT_total_elom_CHILD to PD_4_MV_composite_mn
dtypes: category(132), float32(4)
memory usage: 1.8 MB
